docs/integrations/github/discovery.md

```diff
@@ -80,6 +80,22 @@ catalog:
         filters: # optional filters
           branch: 'develop' # optional string
           repository: '.*' # optional Regex
+      topicProviderId:
+        organization: 'backstage' # string
+        catalogPath: '/catalog-info.yaml' # string
+        filters:
+          branch: 'main' # string
+          repository: '.*' # Regex
+          topic: 'backstage-exclude' # optional string
+      topicProviderInclusionId:
+        organization: 'backstage' # string
+        catalogPath: '/catalog-info.yaml' # string
+        filters:
+          branch: 'main' # string
+          repository: '.*' # Regex
+          topic:
+            name: 'backstage-include' # optional string
```

Comment:
I think we should do something like this instead which would allow us to do both includes and excludes.

```
topic:
  include: ['a']
  exclude: ['b']
```

include would only include repositories with a matching topic.
exclude would exclude any repository with a matching topic.

Would be great to be able to use one of them or in combination

Reply:
Hmm, how would you recommend handling the following case that would be inevitably attempted?
```
topic:
  include: ['a']
  exclude: ['a']
```

I guess it might be reasonable to just throw an error if an item appears inside both `include` and `exclude`

Actually, I'm not sure that having both active would ever make sense anyway? If you're only including repositories with topic `a` then by definition, you'll never be able to exclude `b` because anything that wasn't `a` is never going to be included in the first place

Reply:
There might be cases where you want to include everything with the topic `backstage` but you also want to exclude those that have `legacy` tag for example. That would be

```
topic:
  include: ['backstage']
  exclude: ['legacy']
```

Reply:
Ah ok, I hadn't really thought about applying the exclusion filter to things that pass through the inclusion filter but that makes sense 🙂